### PR TITLE
feat: Add module for Secure Storage Connector

### PR DESCRIPTION
**Pull request by elainaRenee**

### CODE REVIEW

File: modules/app_eks/variables.tf [hunk 1]
```diff
@@ -18,7 +18,7 @@ variable "cluster_version" {
   type        = string
   default     = "1.21"
 }
- 
+
```
Comment (elainaRenee) [inline modules/app_eks/variables.tf:21]: formatting

File: modules/database/main.tf [hunk 2]
```diff
@@ -96,7 +96,7 @@ module "aurora" {
   instance_class = var.instance_class
   instances      = { 1 = {} }
 
-  autoscaling_enabled      = false
+  autoscaling_enabled = false
```
Comment (elainaRenee) [inline modules/database/main.tf:99]: formatting

File: modules/secure_storage_connector/main.tf [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+          "s3:GetBucketCORS",
+          "s3:GetBucketLocation",
+          "s3:GetBucketVersioning"
+        ],
+        "Resource" : [
+          "${module.file_storage.bucket_arn}",
+          "${module.file_storage.bucket_arn}/*",
+        ]
+      }
+    ]
```
Comment (wandb-zacharyblasczyk) [inline modules/secure_storage_connector/main.tf:77]: I had a customer with a statement like this:
```
        {
            "Sid": "DenyEverythingOutsideOfOrg",
            "Effect": "Deny",
            "Principal": "*",
            "Action": "s3:*",
            "Resource": [
                "arn:aws:s3:::org-wandb-file-storage",
                "arn:aws:s3:::org-wandb-file-storage/*"
            ],
            "Condition": {
                "StringNotEquals": {
                    "aws:PrincipalOrgID": [
                        "o-xxxxxxxxxx",
                        "o-xxxxxxxxxx"
                    ]
                }
            }
        }
```
In order to get this working we had to add an additional policy of:
```
        {
            "Sid": "DenyEverythingOutsideOfWandBOrg",
            "Effect": "Deny",
            "Principal": "*",
            "Action": "s3:*",
            "Resource": [
                "arn:aws:s3:::snap-wandb-file-storage",
                "arn:aws:s3:::snap-wandb-file-storage/*"
            ],
            "Condition": {
                "StringNotEquals": {
                    "aws:PrincipalOrgID": [ "o-hn017huuqa" ]
                }
            }
        }
```

We might want to put this behind an `aws_org_statement_enabled` flag. 

LMK your thoughts.

File: modules/secure_storage_connector/README.md [hunk 4]
```diff
@@ -0,0 +1,69 @@
+# Weights & Biases Secure Storage Connector Module
```
Comment (wandb-zacharyblasczyk) [inline modules/secure_storage_connector/README.md:1]: Is this meant to replace the BYOB example? 

Maybe we should update that example to point at this implementation?

**Review by wandb-zacharyblasczyk — COMMENTED** [review]
Overall looks great! Just a few reviews above.

File: modules/secure_storage_connector/outputs.tf [hunk 5]
```diff
@@ -0,0 +1,11 @@
+output "bucket_name" {
+  value = module.file_storage.bucket_name
+}
+
+output "bucket_arn" {
+  value = module.file_storage.bucket_arn
+}
```
Comment (jsbroks) [inline modules/secure_storage_connector/outputs.tf:7]: I've starting perforing this approach, but up to you!
```suggestion
output "bucket" {
  value = module.file_storage
}
```

File: modules/secure_storage_connector/outputs.tf [hunk 6]
```diff
@@ -0,0 +1,11 @@
...
+
+output "bucket_kms_key_arn" {
+  value = var.create_kms_key ? aws_kms_key.key[0].arn : null
+}
```
Comment (jsbroks) [inline modules/secure_storage_connector/outputs.tf:11]: same idea here, return the entire object

**Review by jsbroks — APPROVED** [review]

File: modules/secure_storage_connector/main.tf (hunk 3 above)
Comment (wandb-zacharyblasczyk) [reply modules/secure_storage_connector/main.tf:77]: Sounds good. I will open a ticket.

File: modules/secure_storage_connector/README.md (hunk 4 above)
Comment (elainaRenee) [reply modules/secure_storage_connector/README.md:1]: yeah, they are pretty much the same, but you need to grant access to a different AWS account. I will update the BYOB example in a follow up.

File: modules/secure_storage_connector/outputs.tf (hunk 5 above)
Comment (elainaRenee) [reply modules/secure_storage_connector/outputs.tf]: done

**Comment by jsbroks** [comment]
This PR is included in version 1.11.0 :tada:

**Pull request merged**